lib/surface/compiler/parse_tree_translator.ex

```diff
@@ -109,4 +193,24 @@ defmodule Surface.Compiler.ParseTreeTranslator do
       :ignored_body?
     ])
   end
+
+  defp strip_name_from_tagged_expr_equals!(value, expr_meta) do
+    case Code.string_to_quoted(value) do
```

Comment:
since we have such a limited set of expressions we support `{= ...}` to be used on (and a fairly small set of supported characters for elixir identifiers), would it make sense to just use a regex with a capture group? or is that going to just be harder to maintain with very little benefit?

Reply:
I thought about doing that but for some reason I find it more intuitive using `string_to_quoted` than using regex. I'm not sure why. Maybe it's because I'm always forgetting about some edge case when dealing with regular expressions :)